src/test/java/com/scalar/database/storage/cassandra4driver/BatchHandlerTest.java

```diff
@@ -0,0 +1,386 @@
+            .delete(delete)
+            .build();
+    batch = new BatchHandler(session, handlers);
+  }
+
+  private void configureBehavior() {
+    when(insert.prepare(any(Mutation.class))).thenReturn(prepared);
+    when(insert.bind(any(PreparedStatement.class), any(Mutation.class))).thenReturn(builder);
+    when(update.prepare(any(Mutation.class))).thenReturn(prepared);
+    when(update.bind(any(PreparedStatement.class), any(Mutation.class))).thenReturn(builder);
+    when(delete.prepare(any(Mutation.class))).thenReturn(prepared);
+    when(delete.bind(any(PreparedStatement.class), any(Mutation.class))).thenReturn(builder);
+    when(builder.build()).thenReturn(bound);
+  }
+
+  private List<Mutation> prepareNonConditionalPuts() {
+    Key partitionKey = new Key(new TextValue(ANY_NAME_1, ANY_TEXT_1));
+    Key clusteringKey1 = new Key(new TextValue(ANY_NAME_2, ANY_TEXT_2));
+    Put put1 =
+        new Put(partitionKey, clusteringKey1)
+            .withValue(new IntValue(ANY_NAME_3, ANY_INT_1))
+            .forNamespace(ANY_KEYSPACE_NAME)
+            .forTable(ANY_TABLE_NAME);
+    Key clusteringKey2 = new Key(new TextValue(ANY_NAME_2, ANY_TEXT_3));
+    Put put2 =
+        new Put(partitionKey, clusteringKey2)
+            .withValue(new IntValue(ANY_NAME_3, ANY_INT_1))
+            .forNamespace(ANY_KEYSPACE_NAME)
+            .forTable(ANY_TABLE_NAME);
+    return Arrays.asList(put1, put2);
+  }
+
+  private List<Mutation> prepareConditionalPuts() {
+    List<Mutation> mutations = prepareNonConditionalPuts();
+    mutations.forEach(m -> m.withCondition(new PutIfNotExists()));
+    return mutations;
+  }
+
+  private List<Mutation> prepareMultiPartitionPuts() {
+    Key partitionKey1 = new Key(new TextValue(ANY_NAME_1, ANY_TEXT_1));
+    Key clusteringKey1 = new Key(new TextValue(ANY_NAME_2, ANY_TEXT_2));
+    Put put1 =
+        new Put(partitionKey1, clusteringKey1)
+            .withValue(new IntValue(ANY_NAME_3, ANY_INT_1))
+            .forNamespace(ANY_KEYSPACE_NAME)
+            .forTable(ANY_TABLE_NAME);
+    Key partitionKey2 = new Key(new TextValue(ANY_NAME_4, ANY_TEXT_1));
+    Key clusteringKey2 = new Key(new TextValue(ANY_NAME_2, ANY_TEXT_3));
+    Put put2 =
+        new Put(partitionKey2, clusteringKey2)
+            .withValue(new IntValue(ANY_NAME_3, ANY_INT_1))
+            .forNamespace(ANY_KEYSPACE_NAME)
+            .forTable(ANY_TABLE_NAME);
+    return Arrays.asList(put1, put2);
+  }
+
+  private BatchHandler prepareSpiedBatchHandler() {
+    BatchHandler spy = Mockito.spy(new BatchHandler(session, handlers));
+    return spy;
+  }
+
+  @Test
+  public void handle_CorrectHandlerAndConditionalOperationsGiven_ShouldExecuteProperly()
+      throws ExecutionException {
+    // Arrange
+    configureBehavior();
+    mutations = prepareConditionalPuts();
+    when(session.execute(any(Statement.class))).thenReturn(results);
+    when(results.wasApplied()).thenReturn(true);
+
+    // Act Assert
+    assertThatCode(
```

Comment:
Here, and in other places, do we really have to check that no exceptions are thrown? If an exception is thrown the the test will fail (great!), and if not, the assert part of the test will check the various assertions. Is there something that I am missing?

Reply:
Thank you for the comment!
I think it is useful to check that no exception is thrown to be also compared to other unit tests because it is possible to throw various exceptions from `batch.handle()`.
For example, `handle_EmptyOperationsGiven_ShouldThrowIllegalArgumentException` should be checked to throw an exception. It is pretty sure that this test should NOT throw any exception.